### ISSUE TITLE
Question about trace reading

### ISSUE THREAD

**Issue by mochetts** [equinor/segyio#543]
Hi! I'm building a [segy reader in javascript](https://github.com/OfAshAndFire/segy/blob/feat/mochetts-visualizer/src/lib/parseSegy.ts#L7) and I'm comparing the results of my reader with your reader here.

So far so good except that I get some differences when reading the trace samples on the first trace of [this file](https://github.com/equinor/segyio-notebooks/blob/master/data/basic/3X_75_PR.SGY).

Considering only the first trace (**CDP 1**) of that file...

This is what I get when reading a  **4 bytes float number** using big endian for the first 5 samples **different from 0** (from sample index 188 on):

```
....
188: 44.366153717041016,
189: 103.35108184814453,
190: 56.77102279663086,
191: 117.89983367919922,
192: 40.1371955871582,
...
```

And this is what segyio is reading for the same 5 samples:
```
...
188: 49.46461486816406,
189: 206.70216369628906,
190: 99.08409118652344,
191: 235.79966735839844,
192: 32.54878234863281,
...
```

It seems that some values are exactly doubled by segyio and some others have a different transformation that I can't seem to figure out...

So was wondering... is this a bug on my code? a bug on segyio? Is it that segyio applies some sort of transformation to the numbers?

I'm looking at [this code](https://github.com/equinor/segyio/blob/3044598e6e5dbaa2aff540dfcb6481a1c3c0d242/lib/src/segy.c#L1836) trying to figure out if this is the right place where segyio reads data but it's getting super hard to follow for me. Am I in the right place here?

Thanks!

**Comment by mochetts** [comment]
Some more context... looking at the byte array of the first sample != 0 from the first trace this is what I see:
```
01000010001100010111011011110001
```
This represents exactly `44.366153717041016` in Float32 which is what my javascript parser is reading...

Still puzzled why segyio reads `49.46461486816406`

**Comment by da-wad** [comment]
Hi @mochetts 

You appear to be laboring under the assumption that SEG-Y files exclusively use [IEEE-754 floating point representation](https://en.wikipedia.org/wiki/IEEE_754). This has only actually been possible since [SEG-Y rev.1](https://seg.org/Portals/0/SEG/News%20and%20Resources/Technical%20Standards/seg_y_rev1.pdf)

Unfortunately when you interrogate this particular file...
```
f = segyio.open('segyio-notebooks\\data\\basic\\3X_75_PR.SGY', strict=False)
print(f)
```
... you'll find that what you have is: `float representation: 4-byte IBM float`

which is necessarily the case because:
```
f2.bin[segyio.binfield.BinField.SEGYRevision]
```
returns `0`

I suggest reading more about [IBM floating point](https://en.wikipedia.org/wiki/IBM_hexadecimal_floating-point), and there's [a nice online calculator](http://www.rogerbowler.fr/hfpconv.htm) which should help you figure out how to deal with this in Javascript.

**Comment by mochetts** [comment]
Thanks for the response @da-wad. Much appreciate it.
